### PR TITLE
some overrideDerivation cleanups

### PR DESCRIPTION
**Pull request by infinisil**
These four top-level packages were the only ones that didn't have the
meta.position attribute automagically set. This commit fixes this.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by grahamc** [comment]
FWIW: https://github.com/NixOS/nixpkgs/pull/41165/files#r193124185 but I don't know why.

**Comment by dezgeg** [comment]
Yes, I believe this warning in the linked manual chapter applies to both:
> Warning: Do not use this function in Nixpkgs as it evaluates a Derivation before modifying it, which breaks package abstraction and removes error-checking of function arguments. In addition, this evaluation-per-function application incurs a performance penalty, which can become a problem if many overrides are used. It is only intended for ad-hoc customisation, such as in ~/.config/nixpkgs/config.nix.

File: pkgs/development/interpreters/luajit/default.nix [hunk 1]
```diff
@@ -3,32 +3,23 @@ rec {
 
   luajit = luajit_2_1;
 
-  luajit_2_0 = generic {
+  luajit_2_0 = (generic {
     version = "2.0.5";
     isStable = true;
     sha256 = "0yg9q4q6v028bgh85317ykc9whgxgysp76qzaqgq55y6jy11yjw7";
-  } // {
```
Comment (danbst) [inline pkgs/development/interpreters/luajit/default.nix]: It should be possible to not introduce `overrideAttrs` here:
```
luajit_2_0 = generic {
     version = "2.0.5";
     isStable = true;
     sha256 = "0yg9q4q6v028bgh85317ykc9whgxgysp76qzaqgq55y6jy11yjw7";
     meta = meta // {
      platforms = lib.filter (p: p != "aarch64-linux") meta.platforms;
 };

generic = { ..., meta ? meta }:
 stdenv.mkDerivation rec {
      inherit name version src meta;
....
;
```
`overrideAttrs` doesn't make this generative stuff cleaner IMO.

**Comment by danbst** [comment]
I see no harm in replacing `lib.overrideDerivation` to `.overrideAttrs`, that doesn't make situation "worse". The one that introduces can be rewritten to not introduce.

File: pkgs/development/interpreters/luajit/default.nix (hunk 1 above)
Comment (andir) [reply pkgs/development/interpreters/luajit/default.nix]: I agree. @Infinisil can you adjust it as shown by @danbst?

**Review by andir — CHANGES_REQUESTED** [review]

**Comment by Infinisil** [comment]
Fixed that, but I couldn't use `meta` as the attribute in the top level, because apparently `{ meta ? meta }: ...` tries to use the arguments `meta` for the default value, didn't know that (and it doesn't make any sense for it to be that way)!

**Comment by danbst** [comment]
oh right, forgot func args are implicilty `rec`

**Comment by dezgeg** [comment]
`{ meta ? meta }: ` is pretty nonsensical indeed, but e.g. `{ stdenv, enableFoo ? stdenv.isLinux }:` is pretty useful.

File: pkgs/applications/window-managers/i3/gaps.nix [hunk 2]
```diff
@@ -11,15 +11,14 @@ i3.overrideDerivation (super : rec {
...
 
   postUnpack = ''
       echo -n "${version} (${releaseDate})" > ./i3-${version}/I3_VERSION
   '';
 
   # fatal error: GENERATED_config_enums.h: No such file or directory
   enableParallelBuilding = false;
-}) // {
 
   meta = with stdenv.lib; {
```
Comment (matthewbauer) [inline pkgs/applications/window-managers/i3/gaps.nix]: I don't think meta works with overrideAttrs. Can you confirm that the meta is still being applied? I though you still had to do a union like with overrideDerivation.

Comment (Infinisil) [reply pkgs/applications/window-managers/i3/gaps.nix]: Does indeed work, the main reason I'm doing this PR is to have `meta.position` available. All other meta attributes work as well.

Comment (matthewbauer) [reply pkgs/applications/window-managers/i3/gaps.nix]: Ok looks good then.

**Review by matthewbauer — APPROVED** [review]

**Pull request merged**